### DISCUSSION TITLE
Looking for ideas to solve a problem

### DISCUSSION THREAD

**Discussion by ashraful-i** [mrdbourke/pytorch-deep-learning#641] [General]
My goal is to detect defects in images and highlight those areas. 
I'm looking for ideas/existing solutions to finish this task accurately.
Thank you.

**Comment by KanishkNavale** [comment]
Hi,

You can use simple DL frameworks like Resnet with ROI features predicting masks for defects. If you really need to push the boundaries replace the Resnet with a transformer like MVIT, which gives you a probability heatmap for defects which inturn, highlights the uncertainty.

**Comment by mrdbourke** [comment]
Hi @ashraful-i , 

You'll likely want to look into object detection models, where the defect on the image is the object you'd like to detect.

For example, if you want to detect defects in part manufacturing, you might want to try here: https://blog.roboflow.com/how-to-build-a-defect-detection-system/
